### ISSUE TITLE
SSL: CERTIFICATE_VERIFY_FAILED while trying to download pretrained model within a company that transforms SSL certificates for security purposes

### ISSUE THREAD

**Issue by kim3321** [pytorch/pytorch#91608]
### 🐛 Describe the bug

I currently have difficulty using the code in companies with enhanced security when using external networks.

**from torch.hub import  download_url_to_file
url = 'https://download.openmmlab.com/pretrain/third_party/resnet50_v1c-2cccc1ad.pth'
download_url_to_file(url)**

The following error occurs when I execute the code because the SSL certificate is tampered with by internal security.

  File "C:\Users\user\miniconda3\envs\torch112\lib\site-packages\torch\hub.py", line 727, in load_state_dict_from_url
    download_url_to_file(url, cached_file, hash_prefix, progress=progress)
  File "C:\Users\user\miniconda3\envs\torch112\lib\site-packages\torch\hub.py", line 593, in download_url_to_file
    u = urlopen(req)
  File "C:\Users\user\miniconda3\envs\torch112\lib\urllib\request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "C:\Users\user\miniconda3\envs\torch112\lib\urllib\request.py", line 519, in open
    response = self._open(req, data)
  File "C:\Users\user\miniconda3\envs\torch112\lib\urllib\request.py", line 536, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "C:\Users\user\miniconda3\envs\torch112\lib\urllib\request.py", line 496, in _call_chain
    result = func(*args)
  File "C:\Users\user\miniconda3\envs\torch112\lib\urllib\request.py", line 1391, in https_open
    return self.do_open(http.client.HTTPSConnection, req,
  File "C:\Users\user\miniconda3\envs\torch112\lib\urllib\request.py", line 1351, in do_open
    raise URLError(err)
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: CA certificate key too weak (_ssl.c:997)>

It seems that there was a similar issue as follows #33288
And there is a case of solving the cause in the issue.

**import ssl
ssl._create_default_https_context = ssl._create_unverified_context**

Can I request PR for the code to which this solution is applied?

### Versions

PyTorch version: 1.12.1
Is debug build: False
CUDA used to build PyTorch: 11.6
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Enterprise
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:29:51) [MSC v.1929 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19042-SP0
Is CUDA available: True
CUDA runtime version: 11.6.55
CUDA_MODULE_LOADING set to:
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3090
Nvidia driver version: 516.94
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.23.3
[pip3] torch==1.12.1
[pip3] torchaudio==0.12.1
[pip3] torchvision==0.13.1
[conda] blas                      2.116                       mkl    conda-forge
[conda] blas-devel                3.9.0              16_win64_mkl    conda-forge
[conda] cudatoolkit               11.6.0              hc0ea762_10    conda-forge
[conda] libblas                   3.9.0              16_win64_mkl    conda-forge
[conda] libcblas                  3.9.0              16_win64_mkl    conda-forge
[conda] liblapack                 3.9.0              16_win64_mkl    conda-forge
[conda] liblapacke                3.9.0              16_win64_mkl    conda-forge
[conda] mkl                       2022.1.0           h6a75c08_874    conda-forge
[conda] mkl-devel                 2022.1.0           h57928b3_875    conda-forge
[conda] mkl-include               2022.1.0           h6a75c08_874    conda-forge
[conda] numpy                     1.23.3          py310h4a8f9c9_0    conda-forge
[conda] pytorch                   1.12.1          py3.10_cuda11.6_cudnn8_0    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                0.12.1              py310_cu116    pytorch
[conda] torchvision               0.13.1              py310_cu116    pytorch

cc @nairbv @NicolasHug @vmoens @jdsgomes

**Comment by kim3321** [comment]
@soumith Can you help this issue?

**Comment by vmoens** [comment]
I'm looking into it

**Comment by NicolasHug** [comment]
The issue does look similar to https://github.com/pytorch/pytorch/issues/33288, however I'm not sure there is much we (the pytorch maintainers) can do in this case.

For https://github.com/pytorch/pytorch/issues/33288 we were able to re-new the certificates (https://github.com/pytorch/pytorch/issues/33288#issuecomment-585849493) because the URLs were owned by pytorch, but in this case the offending URL is https://download.openmmlab.com/pretrain/third_party/resnet50_v1c-2cccc1ad.pth which belongs to the `openmmlab` domain. It is owned by https://github.com/open-mmlab/mmcv, not by pytorch. Perhaps you'll have a better chance at getting this fixed by submitting an issue to their repo?

**Comment by kim3321** [comment]
@NicolasHug The example of the website I wrote an example is the web address of openmmlab, as you explained.
 when I used pytorch library, I can't download any pre-trained models because SSL certificates for all **https** websites are modified on-premise environment with for security reasons. In other words, not only the openmmlab website but also all websites accessed by **https** are occured this issue.
Currently, I am executing the program by modifying the source code file **hub.py** directly to avoid this problem.
 I suggest that you modify **hub.py** to select true/false whether to check SSL or not.

**Comment by NicolasHug** [comment]
> Meanwhile, does this help? https://github.com/pytorch/pytorch/issues/33288#issuecomment-954160699

I see that you already had looked into this in your original message (the way it is formatted makes it fairly hard to parse).
Since this suggestion seems to work, I think we can consider this problem solved. I don't think we'll introduce a flag in torchhub to handle that, as it seems to be only affecting some restricted scenarios.

**Comment by kim3321** [comment]
@NicolasHug It's my mistake to press close above.

Some projects allow to choose whether to use the option to troubleshoot SSL problems. When you see an example of the link below,
https://github.com/yt-dlp/yt-dlp
This project offers the following options: 

> --no-check-certificates         Suppress HTTPS certificate validation

If possible, I would appreciate it if you could add a argument to select true/false from **download_url_to_file(url, check_certificate)** function  whether to use ssl or not as the option provided in the above project.

**Comment by Exitare** [comment]
<img width="1249" alt="image" src="https://user-images.githubusercontent.com/13048528/230697807-aee14586-c3d0-4b0e-b3d3-eea4a64203b0.png">
I experience the same issue trying to download the ResNet50 pretrained model.

**Comment by Exitare** [comment]
<img width="1296" alt="image" src="https://user-images.githubusercontent.com/13048528/230697846-34dc8c33-0f9f-4eea-af69-ead837883e0a.png">
I experience the same issue, trying to use the ResNet50 model

**Comment by Exitare** [comment]
<img width="1302" alt="image" src="https://user-images.githubusercontent.com/13048528/230697920-3612a38b-d180-4e8d-9436-12f067f2a1c1.png">
I experience the same issue trying to download ResNet50
